pkg/operator/controller/gatewayclass/servicemeshcontrolplane.go

```diff
@@ -63,12 +61,11 @@ func (r *reconciler) ensureServiceMeshControlPlane(ctx context.Context, gatewayc
 // desiredServiceMeshControlPlane returns the desired servicemeshcontrolplane.
 func desiredServiceMeshControlPlane(name types.NamespacedName, ownerRef metav1.OwnerReference) (*maistrav2.ServiceMeshControlPlane, error) {
 	pilotContainerEnv := map[string]string{
-		"PILOT_ENABLE_GATEWAY_API":                         "true",
-		"PILOT_ENABLE_GATEWAY_API_DEPLOYMENT_CONTROLLER":   "true",
-		"PILOT_ENABLE_GATEWAY_API_STATUS":                  "true",
-		"PILOT_GATEWAY_API_CONTROLLER_NAME":                OpenShiftGatewayClassControllerName,
-		"PILOT_GATEWAY_API_DEFAULT_GATEWAYCLASS":           OpenShiftDefaultGatewayClassName,
-		"PILOT_ENABLE_GATEWAY_API_GATEWAYCLASS_CONTROLLER": "false",
+		"PILOT_ENABLE_GATEWAY_CONTROLLER_MODE":   "true",
+		"PILOT_GATEWAY_API_CONTROLLER_NAME":      OpenShiftGatewayClassControllerName,
+		"PILOT_GATEWAY_API_DEFAULT_GATEWAYCLASS": OpenShiftDefaultGatewayClassName,
+		// Istio will only create the default gateway class if this is true.
+		"PILOT_ENABLE_GATEWAY_API_GATEWAYCLASS_CONTROLLER": "true",
```

Comment:
In [the workflow from the EP](https://github.com/openshift/enhancements/blob/master/enhancements/ingress/gateway-api-with-cluster-ingress-operator.md#workflow-description), the cluster-admin creates the gatewayclass, not Istio.

Reply:
I can update that to say "Istio will only allow ...."

Reply:
For reference: https://github.com/maistra/istio/blob/d8f16bb9d7e2671cf9bf70b1cf4cd314ced01ed0/pilot/pkg/features/pilot.go#L422

Reply:
Hm, so OSSM doesn't reconcile gatewayclasses _at all_ without `PILOT_ENABLE_GATEWAY_API_GATEWAYCLASS_CONTROLLER=true`?  Is that a regression from OSSM 2.4?  Then it seems like are options are (1) no Gateway API support at all or (2) cluster-admin and OSSM clash over ownership of the default gatewayclass object.  Option 2 is better if we have no other option.

Reply:
Yes, it is a regression from OSSM 2.4.  I could ask OSSM to try to fix it in a 2.5 patch, but they don't consider this a bug.  I would like to get this into the next release if possible, if only to let dev preview users play with OSSM 2.5 as-is, but it doesn't add much to the mix in terms of Gateway API.